syntax = "proto3";
package dbinstance;





/*
CreateDBInstance请求
*/
message CreateDBInstanceRequest {
message CreateDbinstance {
message DbClient {
    /*
    客户端IP
    */
    string ip = 1 ;
    /*
    客户端ID
    */
    string id = 2 ;
}
        
    
    /*
    连接客户端
    */
    DbClient dbClient = 1 ;
    /*
    名称
    */
    string name = 2 ;
    /*
    ip
    */
    string ip = 3 ;
    /*
    端口
    */
    int32 port = 4 ;
    /*
    数据库名/服务名
    */
    string dbName = 5 ;
    /*
    连接串
    */
    string connURL = 6 ;
    /*
    默认用户
    */
    string userName = 7 ;
    /*
    默认密码
    */
    string password = 8 ;
    /*
    是否有效
    */
    string isValid = 9 ;
}
        
    
    /*
    数据库服务ID
    */
    string dbServiceId = 1 ;
    /*
    创建数据库实例参数
    */
    CreateDbinstance createDbinstance = 2 ;
}

/*
CreateDBInstance返回
*/
message CreateDBInstanceResponse {
message DbService {
    /*
    数据库服务ID
    */
    string instanceId = 1 ;
    /*
    名称
    */
    string name = 2 ;
    /*
    数据库类型
    */
    string dbType = 3 ;
    /*
    描述
    */
    string desc = 4 ;
}
        
message Client {
    /*
    客户端ID
    */
    string id = 1 ;
    /*
    客户端IP
    */
    string ip = 2 ;
}
        
    
    /*
    数据库服务关系
    */
    repeated DbService dbService = 1 ;
    /*
    客户端关系
    */
    Client client = 2 ;
    /*
    数据库实例ID,自动生成
    */
    string instanceId = 3 ;
    /*
    名称
    */
    string name = 4 ;
    /*
    数据库名/服务名
    */
    string dbName = 5 ;
    /*
    ip
    */
    string ip = 6 ;
    /*
    端口
    */
    int32 port = 7 ;
    /*
    连接串
    */
    string connURL = 8 ;
    /*
    默认用户
    */
    string userName = 9 ;
    /*
    默认密码
    */
    string password = 10 ;
    /*
    是否有效
    */
    string isValid = 11 ;
}

/*
CreateDBInstanceApi返回
*/
message CreateDBInstanceResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    dbinstance.CreateDBInstanceResponse data = 4 ;
}

